searchcore/src/tests/proton/matching/matching_test.cpp

```diff
@@ -528,6 +545,87 @@ TEST("require that re-ranking is performed (multi-threaded)") {
     }
 }
 
+TEST("require that re-ranking is not diverse when not requested to be.") {
+    MyWorld world;
+    world.basicSetup();
+    world.setupSecondPhaseRanking();
+    world.basicResults();
+    SearchRequest::SP request = world.createSimpleRequest("f1", "spread");
+    auto mtf = world.create_mtf(request);
+    auto diversity = mtf->createDiversifier();
+    EXPECT_FALSE(diversity);
+}
+
+using namespace search::fef::indexproperties::matchphase;
+TEST("require that re-ranking is diverse with diversity = 1/1") {
+    MyWorld world;
+    world.basicSetup();
+    world.setupSecondPhaseRanking();
+    world.basicResults();
+    SearchRequest::SP request = world.createSimpleRequest("f1", "spread");
+    auto mtf = world.create_mtf(request);
+    auto & rankProperies = request->propertiesMap.lookupCreate(MapNames::RANK);
+    rankProperies.add(DiversityAttribute::NAME, "a2")
+                 .add(DiversityMinGroups::NAME, "3")
+                 .add(DiversityCutoffStrategy::NAME, "strict");
+    mtf = world.create_mtf(request);
+    auto diversity = mtf->createDiversifier();
+    EXPECT_TRUE(diversity);
+    SearchReply::UP reply = world.performSearch(request, 1);
+    EXPECT_EQUAL(9u, world.matchingStats.docsMatched());
+    EXPECT_EQUAL(9u, world.matchingStats.docsRanked());
+    EXPECT_EQUAL(3u, world.matchingStats.docsReRanked());
+    ASSERT_TRUE(reply->hits.size() == 9u);
+    EXPECT_EQUAL(document::DocumentId("doc::900").getGlobalId(),  reply->hits[0].gid);
+    EXPECT_EQUAL(1800.0, reply->hits[0].metric);
+    EXPECT_EQUAL(document::DocumentId("doc::800").getGlobalId(),  reply->hits[1].gid);
+    EXPECT_EQUAL(1600.0, reply->hits[1].metric);
+    EXPECT_EQUAL(document::DocumentId("doc::700").getGlobalId(),  reply->hits[2].gid);
+    EXPECT_EQUAL(1400.0, reply->hits[2].metric);
+    EXPECT_EQUAL(document::DocumentId("doc::600").getGlobalId(),  reply->hits[3].gid);
+    EXPECT_EQUAL(600.0, reply->hits[3].metric);
+    EXPECT_EQUAL(document::DocumentId("doc::500").getGlobalId(),  reply->hits[4].gid);
+    EXPECT_EQUAL(500.0, reply->hits[4].metric);
+    EXPECT_GREATER(world.matchingStats.matchTimeAvg(), 0.0000001);
+    EXPECT_GREATER(world.matchingStats.rerankTimeAvg(), 0.0000001);
+}
+
+TEST("require that re-ranking is diverse with diversity = 1/10") {
+    MyWorld world;
+    world.basicSetup();
+    world.setupSecondPhaseRanking();
+    world.basicResults();
+    SearchRequest::SP request = world.createSimpleRequest("f1", "spread");
+    auto mtf = world.create_mtf(request);
+    auto diversity = mtf->createDiversifier();
+    EXPECT_FALSE(diversity);
+    auto & rankProperies = request->propertiesMap.lookupCreate(MapNames::RANK);
+    rankProperies.add(DiversityAttribute::NAME, "a3")
+                 .add(DiversityMinGroups::NAME, "3")
+                 .add(DiversityCutoffStrategy::NAME, "strict");
+    mtf = world.create_mtf(request);
+    diversity = mtf->createDiversifier();
+    EXPECT_TRUE(diversity);
```

Comment:
activation check in a separate test would reduce redundant testing...

Reply:
Fixed in #6546